packages/frontend/components/Borrow/TransactionProcessingModal.tsx

```diff
@@ -0,0 +1,311 @@
+import { useEffect, useState } from "react"
+import {
+  Box,
+  Button,
+  Card,
+  CardContent,
+  CircularProgress,
+  Dialog,
+  DialogContent,
+  Grid,
+  Step,
+  StepLabel,
+  Stepper,
+  Typography,
+  useMediaQuery,
+} from "@mui/material"
+import { useTheme, styled } from "@mui/material/styles"
+import StepConnector, {
+  stepConnectorClasses,
+} from "@mui/material/StepConnector"
+import { StepIconProps } from "@mui/material/StepIcon"
+import CloseIcon from "@mui/icons-material/Close"
+import LaunchIcon from "@mui/icons-material/Launch"
+import CheckIcon from "@mui/icons-material/Check"
+import Image from "next/image"
+
+import styles from "../../styles/components/Borrow.module.css"
+import { useTransactionStore } from "../../store/useTransactionStore"
+
+type Step = {
+  label: string
+  description: string
+  link: string | undefined
+}
+
+type TransactionProcessingModalProps = {
+  open: boolean
+  handleClose: (e: {}) => void
+}
+
+const steps: Step[] = [
+  {
+    label: "Deposit 1 ETH on Aave",
+    description: "Ethereum Network",
+    link: "https://ethereum.org/fr/",
+  },
+  {
+    label: "Validate Transaction",
+    description: "Connext bridge",
+    link: "https://www.connext.network/",
+  },
+  {
+    label: "Borrow 900 USDC from Aave",
+    description: "Polygon Network",
+    link: undefined,
+  },
+]
+
+export default function TransactionProcessingModal(
+  props: TransactionProcessingModalProps
+) {
+  const theme = useTheme()
+  const onMobile = useMediaQuery(theme.breakpoints.down("sm"))
+  const [activeStep, setActiveStep] = useState(2)
+  const { transactionStatus, setTransactionStatus } = useTransactionStore(
+    (state) => ({
+      transactionStatus: state.transactionStatus,
+      setTransactionStatus: state.setTransactionStatus,
+    })
```

Comment:
I think you forget to use `shallow` here

https://github.com/pmndrs/zustand#selecting-multiple-state-slices